### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by cdaniluk**

### CODE REVIEW

File: examples/basic/main.tf [hunk 1]
```diff
@@ -1,10 +1,5 @@
+module "cloudberry-infra-bucket" {
+  source = "rhythmictech/cloudberry-bucket/aws"
```
Comment (sblack4) [inline examples/basic/main.tf:2]: does this work without a version?

Comment (sblack4) [reply examples/basic/main.tf:2]: I like to use the examples as my test cases too. I'd like it to be a convention but 🤷

File: main.tf [hunk 2]
```diff
@@ -1,16 +1,151 @@
...
+      }
+    }
+  }
+
+  versioning {
+    enabled = var.versioning_enabled
+  }
+
+  # this cannot be configured programatically via TF, so just ignore it if someone
+  # turned it on administratively.
```
Comment (sblack4) [inline main.tf:69]: Does this sentence need a period? When I write comments I always second guess this. IDK if there's a good argument for either side

File: variables.tf [hunk 3]
```diff
@@ -1,11 +1,67 @@
...
+}
+
+variable "bucket_suffix" {
+  default     = "bucket"
+  description = "Suffix to apply to bucket, which will be created in format [ACCOUNT_ID]-[REGION]-cloudberry-[BUCKET_SUFFIX]. This is not used if `bucket_name` is set to a non-null value."
+  type        = string
+}
+
+variable "cloudberry_account" {
+  default     = "626709717326"
```
Comment (sblack4) [inline variables.tf:14]: Who's account is this? I'm guessing we don't want PII in the PR

**Review by sblack4 — CHANGES_REQUESTED** [review]
I think we want to take the default out of `var. cloudberry_account` before merging

File: variables.tf (hunk 3 above)
Comment (cdaniluk) [reply variables.tf:14]: Account IDs aren't sensitive, and this one is their account, published in their docs. Actually including it is the point, that's something that would help someone who wants to implement this bucket.

**Review by sblack4 — APPROVED** [review]

**Pull request merged**